File: chess-game/chess-game.v3/web/client/chess-app/src/app/services/models/chess-game-dto.ts
export class ChessGameDto {
    game_id: string;
    moves_count: number;
    date: Date;
    name: string;
    state: GameStateDto;
    game_format: GameFormatDto;
    players: PlayersDto;
    board: SquareDto[];
    history: HistoryEntryDto[];
}
export class HistoryEntryDto {
    sequence: number;
    piece: PieceDto;
    from: string;
    to: string;
}
export class GameStateDto {
    turn: string;
    started: boolean;
    finished: boolean;
}
export class GameFormatDto {
    value: string;
    remaining_time: number;
    additional_time: number;
}
export class PlayersDto {
    white_id: string;
    black_id: string;
}
export class PieceDto {
    abbreviation: string;
    piece_id: string;

    public constructor(init?:Partial<PieceDto>) {
        Object.assign(this, init);
    }
}
export class SquareDto {
    square: string;
    piece: PieceDto | null;
    color: string;
    rank: number;
}

File: chess-game/backend/chessapp/application/dtos/chess_game_dto.py
from dataclasses import dataclass
from datetime import datetime, timedelta
from chessapp.domain.kernel.base import BaseResponse


@dataclass(frozen=True, kw_only=True)
class GameStateDto:
    turn: str
    started: bool
    finished: bool

@dataclass(frozen=True, kw_only=True)
class GameFormatDto:
    value: str
    remaining_time: timedelta
    additional_time: timedelta

@dataclass(frozen=True, kw_only=True)
class PlayersDto:
    white_id: str
    black_id: str

@dataclass(frozen=True, kw_only=True)
class ChessGameDto(BaseResponse):
    game_id: str
    date: datetime
    name: str
    state: GameStateDto
    game_format: GameFormatDto
    players: PlayersDto
    board: list
    history: list